driver/native/native.go

```diff
@@ -191,6 +226,26 @@ func (d *Driver) Parse(rctx context.Context, src string) (nodes.Node, error) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
+	if d.state != stateOK && d.state != stateTimeout {
+		return nil, driver.ErrDriverFailure.Wrap(d.lastErr)
+	}
+
+	deadline, _ := ctx.Deadline()
+	if !deadline.IsZero() {
```

Comment:
```suggestion
	if deadline, ok := ctx.Deadline(); ok {
```

Reply:
Is a zero deadline useful? If not, what is the point to check it's existence? Is it to account for weird 3rd-party implementations of the `Context`?

Reply:
No, it's just cosmetic: It gets rid of an explicitly-ignored return value, which is a red flag to the reader (harmless here, but still made me look). And, it narrows the scope of `deadline` which isn't needed outside this block.